client.go

```diff
@@ -1006,20 +1011,20 @@ func (c *Client) lockSessionMonitorChecker(ctx context.Context,
 	go func() {
 		defer c.sessionMonitorCancellations.Delete(monitorName)
 		for {
+			timeUntilDangerZone, err := lock.timeUntilDangerZoneEntered()
```

Comment:
There is a change in logic here that I don't think it is right. It is not a coincidence that the context is checked for cancelation before a routine is triggered. In other words, I don't think this implementation should unconditionally execute this block.

Reply:
Fair enough.  It seemed like an innocuous concession in exchange for cleaner code, but I accept that I was over reaching.  I've changed it so it doesn't alter execution flow, although the code is admittedly clunkier (although no less clear, I think).

I'm less fussed about this function than I am with `acquireLock()` though, the sleep in `acquireLock()` more directly affects user experience (where "user" is an operator hitting ctrl-c ;))

Reply:
I suggest then to revert this one and then we focus on the one that you are most concerned with.